src/build_vehicle_gui.cpp

```diff
@@ -539,6 +540,10 @@ static bool CDECL CargoFilter(const EngineID *eid, const CargoID cid)
 	return (cid == CF_NONE ? refit_mask == 0 : HasBit(refit_mask, cid));
 }
 
+static bool CDECL LocomotiveFilter(EngineID eid) {
+	return Engine::Get(eid)->GetPower() != 0;
+}
+
 static GUIEngineList::FilterFunction * const _filter_funcs[] = {
```

Comment:
Should probably keep this with CargoFilter, rather than splitting it from it

Reply:
Possibly renaming it too, to make it more obvious that it's only for cargo filtering